ETCarsClient.cs

```diff
@@ -88,20 +88,39 @@ private void Client_DataReceived(Sockets.EventArgs.DataReceivedArgs args)
 
             if (dataPoints.Count == 0)
             {
-                string startingData = data.Substring(data.IndexOf("{"));
-                dataPoints.Add(startingData);
+                if (data.Contains("{"))
+                {
+                    string startingData = data.Substring(data.IndexOf("{"));
+                    dataPoints.Add(startingData);
+                }
+                else
+                {
+                    return;
+                }
             }
             else
+            {
                 dataPoints.Add(data);
+            }
 
             if (data.Contains("\r"))
             {
                 string json = String.Join("", dataPoints.ToArray());
                 dataPoints.Clear();
-
-                ETCarsDataContainer deserializedData = JsonConvert.DeserializeObject<ETCarsDataContainer>(json);
-
-                DataReceived?.Invoke(new ETCarsDataReceivedArgs(json, deserializedData.data));
+                
+                int index = json.IndexOf("\r");
+                json = json.Substring(-, index);
```

Comment:
What's going on here?

Reply:
Sometimes the received data can be the full json and a part of the next one and when that's the case the exception is thrown, most of the time I wanted to use the code it threw it.

Reply:
Right, but I meant the substring here has a dash and it says that it fails to compile because of it. What's going on with the dash?

Reply:
Oh, I think it was supposed to be 0, that would make sense (the data to the \r symbol)